### PR TITLE
CRAFTERCMS-2207

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added support for sandbox branch per site configuration

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2207

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java [hunk 1]
```diff
@@ -995,7 +997,11 @@ public boolean createSiteFromBlueprint(String blueprintName, String site) {
             // update site name variable inside config files
             toReturn = helper.updateSitenameConfigVar(site);
         }
-
+/*
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java]: Let's not leave commented out code.

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java [hunk 2]
```diff
@@ -1071,7 +1080,7 @@ public void initialPublish(String site, String environment, String author, Strin
                 // checkout environment branch
                 logger.debug("Checkout environment branch " + environment + " for site " + site);
                 try {
-                    git.checkout().setCreateBranch(true).setForce(true).setStartPoint("master")
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:1074]: Just a comment: this was a bug, no? It didn't honor the config.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java (hunk 2 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:1074]: I don't think it was a bug. That is published repo. The process was to pull from sandbox into published master and from there to deploy content to the environment branch. With this feature, there is an additional obstacle I had to take care off. We can discuss it for further improvements.

**Comment by russdanner** [comment]
@alhambrav to configure this:

In config/studio/site-config.xml add/change:
``` <sandbox-branch>master</sandbox-branch>```

**Pull request merged**